### PR TITLE
[Backport 1.x] Bump BouncyCastle from jdk15on to jdk15to18 (#8247)

### PR DESCRIPTION
**Pull request by willyborankin**
Backport 9856cb7 from #8247

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -118,10 +118,10 @@ dependencies {
   api 'org.apache.maven:maven-model:3.6.2'
   api 'com.networknt:json-schema-validator:1.0.36'
   api "com.fasterxml.jackson.core:jackson-databind:${props.getProperty('jackson_databind')}"
-  api "org.bouncycastle:bcprov-jdk15on:${props.getProperty('bouncycastle')}"
-  api "org.bouncycastle:bcpkix-jdk15on:${props.getProperty('bouncycastle')}"
-  api "org.bouncycastle:bcpg-jdk15on:${props.getProperty('bouncycastle')}"
-  api "org.bouncycastle:bcutil-jdk15on:${props.getProperty('bouncycastle')}"
+  api "org.bouncycastle:bcprov-jdk15to18:${props.getProperty('bouncycastle')}"
```
Comment (reta) [inline buildSrc/build.gradle:121]: So this is what we get:
 - the `org.bouncycastle:*`  overrides were needed for `api 'com.netflix.nebula:gradle-info-plugin:8.2.0'
 - those are explicitly using  `xxx-jdk15on`

Comment (reta) [reply buildSrc/build.gradle:121]: This is how the dependencies look like now

Comment (willyborankin) [reply buildSrc/build.gradle:121]: Ok I will change it back to 1.70 for jdk15on

Comment (willyborankin) [reply buildSrc/build.gradle]: @reta done

File: test/fixtures/hdfs-fixture/build.gradle [hunk 2]
```diff
@@ -47,7 +47,7 @@ dependencies {
   api "org.apache.logging.log4j:log4j-core:${versions.log4j}"
   api "io.netty:netty-all:${versions.netty}"
   api 'com.google.code.gson:gson:2.9.0'
-  api "org.bouncycastle:bcpkix-jdk15on:${versions.bouncycastle}"
+  api "org.bouncycastle:bcpkix-jdk15to18:${versions.bouncycastle}"
```
Comment (reta) [inline test/fixtures/hdfs-fixture/build.gradle:50]: Here the same issue:

```
org.bouncycastle:bcpkix-jdk15on:1.68
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcpkix-jdk15on:1.68
\--- org.apache.hadoop:hadoop-yarn-server-web-proxy:3.3.5
     +--- org.apache.hadoop:hadoop-minicluster:3.3.5
     |    \--- runtimeClasspath
     +--- org.apache.hadoop:hadoop-mapreduce-client-app:3.3.5
     |    +--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)
     |    \--- org.apache.hadoop:hadoop-mapreduce-client-hs:3.3.5
     |         \--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)
     \--- org.apache.hadoop:hadoop-yarn-server-resourcemanager:3.3.5
          \--- org.apache.hadoop:hadoop-yarn-server-tests:3.3.5
               \--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)

org.bouncycastle:bcpkix-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcpkix-jdk15to18:1.75
\--- runtimeClasspath

org.bouncycastle:bcprov-jdk15on:1.68
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcprov-jdk15on:1.68
+--- org.apache.hadoop:hadoop-yarn-server-web-proxy:3.3.5
|    +--- org.apache.hadoop:hadoop-minicluster:3.3.5
|    |    \--- runtimeClasspath
|    +--- org.apache.hadoop:hadoop-mapreduce-client-app:3.3.5
|    |    +--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)
|    |    \--- org.apache.hadoop:hadoop-mapreduce-client-hs:3.3.5
|    |         \--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)
|    \--- org.apache.hadoop:hadoop-yarn-server-resourcemanager:3.3.5
|         \--- org.apache.hadoop:hadoop-yarn-server-tests:3.3.5
|              \--- org.apache.hadoop:hadoop-minicluster:3.3.5 (*)
\--- org.bouncycastle:bcpkix-jdk15on:1.68
     \--- org.apache.hadoop:hadoop-yarn-server-web-proxy:3.3.5 (*)

org.bouncycastle:bcprov-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcprov-jdk15to18:1.75
+--- org.bouncycastle:bcpkix-jdk15to18:1.75
|    \--- runtimeClasspath
\--- org.bouncycastle:bcutil-jdk15to18:1.75
     \--- org.bouncycastle:bcpkix-jdk15to18:1.75 (*)

org.bouncycastle:bcutil-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcutil-jdk15to18:1.75
\--- org.bouncycastle:bcpkix-jdk15to18:1.75
     \--- runtimeClasspath

```

File: buildSrc/build.gradle (hunk 1 above)
Comment (reta) [reply buildSrc/build.gradle]: As we agreed, let's do substitution instead:

```
  modules {
    module("org.bouncycastle:bcprov-jdk15on") {
      replacedBy("org.bouncycastle:bcprov-jdk15to18", "the jdk15on artifacts are not supported anymore")
    }
    module("org.bouncycastle:bcpkix-jdk15on") {
      replacedBy("org.bouncycastle:bcpkix-jdk15to18", "the jdk15on artifacts are not supported anymore")
    }
    module("org.bouncycastle:bcpg-jdk15on") {
      replacedBy("org.bouncycastle:bcpg-jdk15to18", "the jdk15on artifacts are not supported anymore")
    }
    module("org.bouncycastle:bcutil-jdk15on") {
      replacedBy("org.bouncycastle:bcutil-jdk15to18", "the jdk15on artifacts are not supported anymore")
    }
  }
```

File: test/fixtures/hdfs-fixture/build.gradle (hunk 2 above)
Comment (reta) [reply test/fixtures/hdfs-fixture/build.gradle:50]: As we discussed, let's try to exclude `org.bouncycastle` (this is test deps)

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -100,6 +100,20 @@ repositories {
 
 dependencies {
 
+  modules {
```
Comment (reta) [inline buildSrc/build.gradle:103]:
```
> Task :build-tools:dependencyInsight
org.bouncycastle:bcpg-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Selected by rule : the jdk15on artifacts are not supported anymore

org.bouncycastle:bcpg-jdk15on:1.64 -> org.bouncycastle:bcpg-jdk15to18:1.75
\--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r
     \--- com.netflix.nebula:gradle-info-plugin:8.2.0
          \--- runtimeClasspath

org.bouncycastle:bcpg-jdk15to18:1.75
\--- runtimeClasspath

org.bouncycastle:bcpkix-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Selected by rule : the jdk15on artifacts are not supported anymore

org.bouncycastle:bcpkix-jdk15on:1.64 -> org.bouncycastle:bcpkix-jdk15to18:1.75
\--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r
     \--- com.netflix.nebula:gradle-info-plugin:8.2.0
          \--- runtimeClasspath

org.bouncycastle:bcpkix-jdk15to18:1.75
\--- runtimeClasspath

org.bouncycastle:bcprov-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - Selected by rule : the jdk15on artifacts are not supported anymore

org.bouncycastle:bcprov-jdk15on:1.64 -> org.bouncycastle:bcprov-jdk15to18:1.75
\--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r
     \--- com.netflix.nebula:gradle-info-plugin:8.2.0
          \--- runtimeClasspath

org.bouncycastle:bcprov-jdk15to18:1.75
+--- runtimeClasspath
+--- org.bouncycastle:bcpg-jdk15to18:1.75
|    +--- runtimeClasspath
|    \--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r (requested org.bouncycastle:bcpg-jdk15on:1.64)
|         \--- com.netflix.nebula:gradle-info-plugin:8.2.0
|              \--- runtimeClasspath
+--- org.bouncycastle:bcpkix-jdk15to18:1.75
|    +--- runtimeClasspath
|    \--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r (requested org.bouncycastle:bcpkix-jdk15on:1.64) (*)
\--- org.bouncycastle:bcutil-jdk15to18:1.75
     +--- runtimeClasspath
     \--- org.bouncycastle:bcpkix-jdk15to18:1.75 (*)

org.bouncycastle:bcutil-jdk15to18:1.75
   variant "runtime" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-runtime
      org.gradle.libraryelements     = jar
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]

org.bouncycastle:bcutil-jdk15to18:1.75
+--- runtimeClasspath
\--- org.bouncycastle:bcpkix-jdk15to18:1.75
     +--- runtimeClasspath
     \--- org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r (requested org.bouncycastle:bcpkix-jdk15on:1.64)
          \--- com.netflix.nebula:gradle-info-plugin:8.2.0
               \--- runtimeClasspath
```

**Review by reta — APPROVED** [review]

**Pull request merged**